### PR TITLE
Add support for specifying logging configuration for kubelet

### LINKED ISSUE #1166: Allow logging configuration for Kubelet
Right now, with containerd, uses 10Mi is used as default size. And for Docker, it seems to be 100Mi (configured in /etc/docker/daemon.json).

Increase container log max size to 100Mi by default and allow the user to configure the size too

### PR DESCRIPTION
**Pull request by nerdeveloper**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>` format, will close the issue(s) when PR gets merged)*:
Fixes #1166

**Special notes for your reviewer**:

**Optional Release Note**:

```release-note
* Default container log size increased to 100mi for both docker and containerd
* Support for specifying logging configuration using annotations
```

### CODE REVIEW

File: pkg/userdata/helper/kubelet.go [hunk 1]
```diff
@@ -203,6 +205,7 @@ func kubeletConfiguration(clusterDomain string, clusterDNS []net.IP, featureGate
 		EvictionHard:          map[string]string{"memory.available": "100Mi", "nodefs.available": "10%", "nodefs.inodesFree": "5%", "imagefs.available": "15%"},
 		VolumePluginDir:       "/var/lib/kubelet/volumeplugins",
 		TLSCipherSuites:       kubeletTLSCipherSuites,
+		ContainerLogMaxSize:   "100Mi",
```
Comment (kron4eg) [inline pkg/userdata/helper/kubelet.go:208]: should also be a constant

**Review by kron4eg — APPROVED** [review]
/approve

**Review by kron4eg — APPROVED** [review]

**Review by ahmedwaleedmalik — APPROVED** [review]
/lgtm

**Review by ahmedwaleedmalik — COMMENTED** [review]
/approve

**Pull request merged**